src/components/ContentPreview/ContentPreview.js

```diff
@@ -1078,73 +1083,89 @@ class ContentPreview extends PureComponent<Props, State> {
+                                onPrint={this.print}
+                                canDownload={this.canDownload()}
+                                onDownload={this.download}
+                                contentOpenWithProps={contentOpenWithProps}
+                                canAnnotate={this.canAnnotate()}
+                            />
+                        )}
+                        <div className="bcpr-body">
+                            <div className="bcpr-container" onMouseMove={this.onMouseMove} ref={this.containerRef}>
+                                {file ? (
+                                    <Measure bounds onResize={this.onResize}>
+                                        {({ measureRef: previewRef }) => (
+                                            <div ref={previewRef} className="bcpr-content" />
+                                        )}
+                                    </Measure>
+                                ) : (
+                                    <div className="bcpr-loading-wrapper">
+                                        <PreviewLoading
+                                            isLoading={!isFileError}
+                                            loadingIndicatorProps={{
+                                                size: 'large',
+                                            }}
+                                        />
+                                    </div>
+                                )}
+                                {hasLeftNavigation && (
+                                    <PlainButton
+                                        type="button"
+                                        className="bcpr-navigate-left"
+                                        onClick={this.navigateLeft}
+                                    >
+                                        <IconNavigateLeft />
+                                    </PlainButton>
+                                )}
+                                {hasRightNavigation && (
+                                    <PlainButton
+                                        type="button"
+                                        className="bcpr-navigate-right"
+                                        onClick={this.navigateRight}
+                                    >
+                                        <IconNavigateRight />
+                                    </PlainButton>
+                                )}
+                            </div>
+                            {file && (
+                                <ContentSidebar
+                                    {...contentSidebarProps}
+                                    isLarge={isLarge}
+                                    apiHost={apiHost}
+                                    token={token}
+                                    cache={this.api.getCache()}
+                                    features={contentSidebarProps.features || features}
```

Comment:
ContentSidebar is a top-level export from BUIE, so it also needs to have a FeatureProvider. This passes the parent feature options down when it used in the Preview.

Reply:
Is it possible for `FeatureProvider` to check if an existing `FeatureContext` value already exists and, if so, use that? That should allow the value to cascade without the need to explicitly `OR` check each time.

Reply:
My preference to solve this issue would be to not import top-level exports into each other

Reply:
I agree that the current top-level cascade isn't ideal, but it's a much larger refactor to change that at the moment. The code snippet you posted looks like it's worth trying.

Reply:
OK, refactoring/extending the feature framework seems like it can be a separate PR